plugins/database/quota/src/main/java/org/apache/cloudstack/api/response/QuotaResponseBuilderImpl.java

```diff
@@ -342,76 +371,177 @@ public int compare(QuotaBalanceVO o1, QuotaBalanceVO o2) {
     }
 
     @Override
-    public QuotaStatementResponse createQuotaStatementResponse(final List<QuotaUsageVO> quotaUsage) {
-        if (quotaUsage == null || quotaUsage.isEmpty()) {
-            throw new InvalidParameterValueException("There is no usage data found for period mentioned.");
+    public QuotaStatementResponse createQuotaStatementResponse(final List<QuotaUsageJoinVO> quotaUsages, QuotaStatementCmd cmd) {
+        if (CollectionUtils.isEmpty(quotaUsages)) {
+            throw new InvalidParameterValueException(String.format("There is no usage data for parameters [%s].", ReflectionToStringBuilderUtils.reflectOnlySelectedFields(cmd,
+                    "accountName", "accountId", "domainId", "startDate", "endDate", "type", "showDetails")));
         }
+        logger.debug("Creating quota statement from [{}] usage records for parameters [{}].", quotaUsages.size(),
+                ReflectionToStringBuilderUtils.reflectOnlySelectedFields(cmd, "accountName", "accountId", "domainId", "startDate", "endDate", "type", "showDetails"));
```

Comment:
this seems the reuse of the same array of strings. Not for this PR but maybe we should allow String[] as parameter for `ReflectionToStringBuilderUtils.reflectOnlySelectedFields`.  Or allow some kind of predefined default per class to be registered.